src/js.c

```diff
@@ -2764,12 +2764,12 @@ _subscribe(natsSubscription **new_sub, jsCtx *js, const char *subject, const cha
            natsMsgHandler usrCB, void *usrCBClosure, bool isPullMode,
            jsOptions *jsOpts, jsSubOptions *opts, jsErrCode *errCode)
 {
+    natsStatus s = NATS_OK;
     const char *subjects[] = {subject};
+    int numSubjects = nats_IsStringEmpty(subject) ? 0 : 1;
 
-    if (nats_IsStringEmpty(subject))
-        return _subscribeMulti(new_sub, js, NULL, 0, pullDurable, usrCB, usrCBClosure, isPullMode, jsOpts, opts, errCode);
-
-    return _subscribeMulti(new_sub, js, subjects, 1, pullDurable, usrCB, usrCBClosure, isPullMode, jsOpts, opts, errCode);
+    s = _subscribeMulti(new_sub, js, subjects, numSubjects, pullDurable, usrCB, usrCBClosure, isPullMode, jsOpts, opts, errCode);
```

Comment:
Not sure... looks like there is a case in _subscribeMulti where if numSubjects == 0 we would still set cfg->FilterSubjects to subjects (which in this case would not be NULL). But cfg->FilterSubjectsLen will be correctly set to 0, which I believe all other code use the len before accessing the FilterSubjects field. But to be safe, I think if numSubjects == 0, we should have subjects set to NULL. If that is not "clean" to do, then revert to check of subject being empty and invoking with different values.

Reply:
Yes, that was my understanding as well, but cleared it out now as you suggested.